device_doctor/bin/main.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+
+import 'package:device_doctor/src/config.dart';
+
+const String actionFlag = 'action';
+const String deviceOSFlag = 'deviceOS';
+const String helpFlag = 'help';
+
+/// These values will be initialized in `_checkArgs` function,
+/// and used in `main` function.
+String _action;
+String _deviceOS;
+
+/// Manage healthcheck and recovery for devices.
+///
+/// Usage:
+/// dart main.dart --action <healthcheck|recovery> --deviceOS <android|ios>
+Future<void> main(List<String> args) async {
+  final ArgParser parser = ArgParser();
+  parser
+    ..addOption('$actionFlag')
+    ..addOption('$deviceOSFlag')
+    ..addFlag('$helpFlag');
+
+  if (!_checkArgs(parser, args)) {
+    stdout.write('\nRequired flags:\n'
+        '--$actionFlag Supported actions are healthcheck and recovery.\n'
+        '--$deviceOSFlag Supported device OS: android and ios.\n');
+    exit(1);
+  }
+
+  Config.initialize(_deviceOS);
+
+  // Healthcheck and recovery will be implemented here. PR to be continued.
```

Comment:
Is there an issue we can track to?

Reply:
Added https://github.com/flutter/flutter/issues/66193